pkg/kubelet/kubelet_pods.go

```diff
@@ -1570,16 +1577,14 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 	return *s
 }
 
-// convertStatusToAPIStatus creates an api PodStatus for the given pod from
-// the given internal pod status.  It is purely transformative and does not
-// alter the kubelet state at all.
-func (kl *Kubelet) convertStatusToAPIStatus(pod *v1.Pod, podStatus *kubecontainer.PodStatus) *v1.PodStatus {
-	var apiPodStatus v1.PodStatus
-
-	// The runtime pod status may have an arbitrary number of IPs, in an arbitrary
-	// order. Pick out the first returned IP of the same IP family as the node IP
-	// first, followed by the first IP of the opposite IP family (if any).
-	podIPs := make([]v1.PodIP, 0, len(podStatus.IPs))
+// sortPodIPs return the PodIPs sorted and truncated by the cluster IP family preference.
+// The runtime pod status may have an arbitrary number of IPs, in an arbitrary order.
+// PodIPs are obtained by: func (m *kubeGenericRuntimeManager) determinePodSandboxIPs()
+// Pick out the first returned IP of the same IP family as the node IP
+// first, followed by the first IP of the opposite IP family (if any)
+// and use them for the Pod.Status.PodIPs and the Downward API environment variables
+func (kl *Kubelet) sortPodIPs(podIPs []string) []string {
+	ips := make([]string, 0, len(podIPs))
```

Comment:
does this capacity ever need to be > 2?  Why not just use literal 2?

Reply:
it can be 1

Reply:
ah, I got it now